rules/php80/tests/Rector/If_/NullsafeOperatorRector/Fixture/fixture_next_is_property.php.inc

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Rector\Php80\Tests\Rector\If_\NullsafeOperatorRector\Fixture;
+
+class SomeClassNextProperty
+{
+    public function f($o)
+    {
+        $o2 = $o->property1;
+        if ($o2 === null) {
+            return null;
+        }
+
+        return $o2->property2;
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Php80\Tests\Rector\If_\NullsafeOperatorRector\Fixture;
+
+class SomeClassNextProperty
+{
+    public function f($o)
+    {
+        return $o?->property1?->property2;
```

Comment:
This seems wrong. Why do we need `?->` for `property1`?

Reply:
That's needed to check if that's null, return early, else that uses re-assign with new object and call other property by new new object, ref https://3v4l.org/eZGoF

Reply:
But in the original code we're calling `$o->property1` without checking for `null`, these two code snippets will not actually behave the same. One throws a warning (and possibly an exception in the future), the other doesn't.

https://3v4l.org/lGM7B

Reply:
When user write `$o2 = $o->property1`,  it is user responsibility to know that `$property1` exists. On the fixture example, it is assigned to `$o2`, next `$o2` is check against null via `if` statement to return null early if its result is null, otherwise, call `property2` on by $o2 as new object.

Your example is invalid as $o is a null, not object.

Reply:
Correct me if I'm wrong, but Rector shouldn't alter the behavior of code (at least in this case).

```php
$o2 = $o->property1;
if ($o2 === null) {
    return null;
}
return $o2->property2;
=====
return $o?->property1?->property2;
```

Those two code snippets are not semantically equivalent. The former will warn when `$o` is `null` but the latter will. The code you actually want is `return $o->property1?->property2;` (allow `property1` to be null but not `$o`).

> Your example is invalid as $o is a null, not object.

That's kind of the point, `$o?->property1` will check if `$o` is `null` which it will never be. Thus in this case the normal `->` operator should be used. If for some reason `$o` would actually be `null` the latter would just ignore the `null` value. This might not be intended by the user.

Reply:
No, this rule is interpreting what user wrote was already correct. So, `$o->property1` will returns 2 things: null, or object that has `property2`.

Reply:
Side note: I'm the author and implementer of the nullsafe RFC.

> So, $o->property1 will returns 2 things: null, or object that has property2.

Yes, but the point is that you don't need `?->` before `property1`, just `property2`. Note that the nullsafe RFC, `?->` will check for `null` on the left hand side of the operator, not the right hand side.

https://3v4l.org/fr6qQ

As you can see, by making `->` into a `?->` you'll convert an exception to a `null` value for something that the user might have never considered. This is inherently unsafe. This obviously can't happen if you type hint the parameter properly but that's not necessarily always the case (it isn't in the test case).

Anyway, let's see what @TomasVotruba thinks about this :slightly_smiling_face:

Reply:
Ok, I got your point to remove `?->` in very start property/method call. Next `->` on position 2, 3, and so on will still needed. 

Probably another PR for improvement ;)

Reply:
@samsonasik Cool :slightly_smiling_face: Thank you for your contributions!